### PR TITLE
Issue #29: Solr & DB support

### PR DESCRIPTION
**Pull request by ekes**

### CODE REVIEW

File: modules/localgov_search_solr/config/install/search_api.server.localgov_sitewide_solr.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+backend: search_api_solr
+backend_config:
+  retrieve_data: false
+  highlight_data: false
+  site_hash: false
+  server_prefix: ''
+  domain: generic
+  environment: default
+  connector: standard
+  connector_config:
```
Comment (ekes) [inline modules/localgov_search_solr/config/install/search_api.server.localgov_sitewide_solr.yml:38]: Note:

This is installing with the configuration that lando will ship with.
It does _not_ cause an error on installation if the server is not there.
It does show a red mark against the server as unavailable. Clearly requiring editing for the correct configuration.

Don't think, with some documentation, we can do better than that, for a module install.

**Comment by ekes** [comment]
Open question about where to put the configuration for the index specific to the server https://github.com/localgovdrupal/localgov_search/issues/29#issuecomment-1175221402

**Comment by ekes** [comment]
Handy lando configuration at the project level https://github.com/localgovdrupal/localgov_project/tree/feature/2.x/localgov_search-29

**Comment by finnlewis** [comment]
There are integration tests on this.

Currently the tests fail due to missing dependency on search_api_solr

Can we add search_api_solr to require-dev of the project composer, then change our testing install script to include requiire-dev?

**Comment by finnlewis** [comment]
@ekes will take a look, but interested to know if @stephen-cox has any thoughts!

**Comment by andybroomfield** [comment]
Seeing the following error when trying to view the the search sitewide server.
>  Symfony\Component\Routing\Exception\RouteNotFoundException: Route "solr_configset.config_zip" does not exist. in Drupal\Core\Routing\RouteProvider->getRouteByName() (line 206 of core/lib/Drupal/Core/Routing/RouteProvider.php).

**Comment by andybroomfield** [comment]
I'm also seeing the following when checking this out after enabling localgov_demo and then localgov_search_solr
> TypeError: Solarium\Core\Client\Client::__construct(): Argument #2 ($eventDispatcher) must be of type Psr\EventDispatcher\EventDispatcherInterface, Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher given, called in /app/web/modules/contrib/search_api_solr/src/SolrConnector/SolrConnectorPluginBase.php on line 367 in Solarium\Core\Client\Client->__construct() (line 293 of /app/vendor/solarium/solarium/src/Core/Client/Client.php).

This is in the UI and not via Drush which seems fine.

**Pull request merged**